src/DI.php

```diff
@@ -218,10 +218,30 @@ public static function storageManager()
 		return self::$dice->create(Core\Storage\Repository\StorageManager::class);
 	}
 
+	/**
+	 * @return \Friendica\Core\System
+	 */
+	public static function system()
+	{
+		return self::$dice->create(Core\System::class);
+	}
+
 	//
 	// "LoggerInterface" instances
 	//
 
+	/**
+	 * Flushes the Logger instance, so the factory is called again
+	 * (creates a new id and retrieves the current PID)
+	 */
+	public static function flushLogger()
+	{
+		$flushDice = self::$dice
+			->addRule(LoggerInterface::class, self::$dice->getRule(LoggerInterface::class))
+			->addRule('$devLogger', self::$dice->getRule('$devLogger'));
+		static::init($flushDice);
```

Comment:
Why do we need to re-add the same Logger rules?

Reply:
Because this wipes the private instance cache of Dice for these classes, so inside a child, the logger will get build again